modules/tide_content_collection/src/Plugin/Field/FieldWidget/ContentCollectionConfigurationWidget.php

```diff
@@ -0,0 +1,873 @@
+        '#suffix' => '</div>',
+        '#allowed_tags' => ['div'],
+      ];
+      return $element;
+    }
+
+    $json = $element['value']['#default_value'];
+    $json_object = [];
+    if (!empty($json)) {
+      $json_object = json_decode($json, TRUE);
+      if ($json_object === NULL) {
+        $json_object = [];
+      }
+    }
+
+    $element['title'] = [
+      '#title' => $this->t('Title'),
+      '#type' => 'textfield',
+      '#description' => 'Title displayed above results.',
+      '#default_value' => $json_object['title'] ?? '',
+      '#weight' => 1,
+    ];
+
+    $element['description'] = [
+      '#title' => $this->t('Description'),
+      '#type' => 'textarea',
+      '#description' => 'Description displayed above the results',
+      '#default_value' => $json_object['description'] ?? '',
+      '#weight' => 2,
+    ];
+
+    if (!empty($settings['tabs']['content']['enable_call_to_action']) && $settings['tabs']['content']['enable_call_to_action']) {
+      $element['callToAction'] = [
+        '#type' => 'details',
+        '#title' => $this->t('Call To Action'),
+        '#description' => 'A link to another page.',
+        '#open' => TRUE,
+        '#weight' => 3,
+      ];
+      $element['callToAction']['text'] = [
+        '#type'  => 'textfield',
+        '#title'  => $this->t('Text'),
+        '#default_value' => $json_object['callToAction']['text'] ?? '',
+        '#description' => $this->t('Display text of the link.'),
+      ];
+      $element['callToAction']['url'] = [
+        '#type' => 'url',
+        '#title'  => $this->t('Url'),
+        '#default_value' => $json_object['callToAction']['url'] ?? '',
+      ];
+    }
+
+    $configuration = $items[$delta]->configuration ?? [];
+
+    $element['#attached']['library'][] = 'field_group/formatter.horizontal_tabs';
+
+    $element['tabs'] = [
+      '#type' => 'horizontal_tabs',
+      '#tree' => TRUE,
+      '#weight' => 4,
+      '#group_name' => 'tabs',
+    ];
+
+    $this->buildContentTab($items, $delta, $element, $form, $form_state, $configuration, $json_object);
+    $this->buildLayoutTab($items, $delta, $element, $form, $form_state, $configuration, $json_object);
+
+    return $element;
+  }
+
+  /**
+   * Build Content Tab.
+   *
+   * @param \Drupal\Core\Field\FieldItemListInterface $items
+   *   Field items.
+   * @param int $delta
+   *   The current delta.
+   * @param array $element
+   *   The element.
+   * @param array $form
+   *   The form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The form state.
+   * @param array $configuration
+   *   The YAML configuration of the listing.
+   * @param array $json_object
+   *   The json_object of the listing.
+   */
+  protected function buildContentTab(FieldItemListInterface $items, $delta, array &$element, array &$form, FormStateInterface $form_state, array $configuration = NULL, array $json_object = NULL) {
+    $settings = $this->getSettings();
```

Comment:
the method body is too big. Please break it down for readability.

Reply:
Methods broken down into sub sections.